client/modules/sheaf-core/src/print-tree.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file Exports the `printTree` (AST stringify) function for Sheaf.
+ */
+
+import { Input, NodeType, stringInput, Tree, TreeCursor } from "lezer-tree"
+
+// credit to: https://discuss.codemirror.net/u/grayen/summary
+function focusedNode(
+  cursor: TreeCursor
+): { readonly type: NodeType; readonly from: number; readonly to: number } {
+  const { type, from, to } = cursor
+  return { type, from, to }
+}
+// credit to: https://discuss.codemirror.net/u/grayen/summary
+export function printTree(
+  tree: Tree,
+  input: Input | string,
+  options: { from?: number; to?: number; start?: number; includeParents?: boolean } = {}
+): string {
+  const cursor = tree.cursor()
+  if (typeof input === "string") input = stringInput(input)
+  const { from = 0, to = input.length, start = 0, includeParents = false } = options
+  let output = ""
+  const prefixes: string[] = []
+  for (;;) {
```

Comment:
The convention in JS/TS isn't `while (true) {`?

Reply:
It's a bit of a conflict - and this isn't directly my code. CodeMirror internals use `for(;;)`, while yeah, I would personally use `while (true)`. I think there might be a lint rule for that?

Reply:
Okay, I've looked into this, and the result is confusing to me. Apparently, eslint's [no-constant-expression](https://eslint.org/docs/rules/no-constant-condition) rule, enabled by default, actually _disallows_ `while (true)`, while accepting `for (;;)`. If I had a guess why that is, it's because `for (;;)` is a bit more... in your face and obvious? There doesn't seem to be a way to configure it to work the other way around, which is strange.